src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -164,4 +188,22 @@ private function getTimezone(array $context)
 
         return $dateTimeZone instanceof \DateTimeZone ? $dateTimeZone : new \DateTimeZone($dateTimeZone);
     }
+
+    private function isPreserveContextTimezone(array $context): bool
+    {
+        //Version 5.0 of Symfony will always preserve the context timezone.
+        if (class_exists('\Symfony\Component\HttpKernel\Kernel') && version_compare(Kernel::VERSION, '5.0.00', '>=')) {
+            return true ;
+        }
+
+        if (!isset($context[self::PRESERVE_CONTEXT_TIMEZONE]) && !isset($this->defaultContext[self::PRESERVE_CONTEXT_TIMEZONE])) {
+            @trigger_error('Not setting the boolean "PRESERVE_CONTEXT_TIMEZONE" flag is deprecated. Set the flag to "true" to apply the context timezone consistently, otherwise setting the flag to "false" will preserve default PHP behavior.', E_USER_DEPRECATED);
+        }
+
+        if (!isset($context[self::PRESERVE_CONTEXT_TIMEZONE])) {
```

Comment:
All the lines below can be simplified using:

```php
return $context[self::PRESERVE_CONTEXT_TIMEZONE] ?? $this->defaultContext[self::PRESERVE_CONTEXT_TIMEZONE] ?? false;
```

right?

Reply:
My system throws errors with this code as the array does not necessary contain the self::PRESERVE_CONTEXT_TIMEZONE options, so my answer on this one is No.

Reply:
That's the point of `??` though, it shouldn't throw errors if the key is undefined.